lib/index.js

```diff
@@ -974,3 +984,23 @@ exports.shallow = function (source) {
 
     return target;
 };
+
+
+exports.promiseWrap = function (bind, method, args, transform) {
+
+    return new Promise((resolve, reject) => {
+
+        const callback = function (error, result) {
+
+            if (Array.isArray(transform)) {
+                result = exports.transform(Array.from(arguments), transform);
+                error instanceof Error ? reject(result) : resolve(result);
```

Comment:
I don't follow you. This use case is just passing all parameters because it doesn't care about their values, but it's unlikely we pass errors and values and care about both in the end.